File: inp_generation/Ti_alpha-beta_v2/Euler_to_GSH.m
ns = 50;
el = 21;

load euler_50val.mat

euler_GSH = zeros(el^3,10,ns);

for sn = 1 : ns
    for k = 1 : el^3
        euler_GSH(k,:,sn)= GSH_Cubic_Triclinic(euler(k,sn,1),euler(k,sn,2),euler(k,sn,3));
    end
    sn
end

euler_GSH = permute(euler_GSH,[1 3 2]);

save('euler_cubic_GSH_50val','euler_GSH');

File: fip_collab/2015_01_12_strain_alpha/euler_to_gsh.py
# -*- coding: utf-8 -*-
"""
Created on Tue Jul 15 11:45:46 2014

@author: nhpnp3
"""

import numpy as np
import GSH_func as gsh
import functions as rr
import time

def euler_to_gsh(el,H,ns,set_id,step,wrt_file):

    start = time.time()
    
    euler = np.load('euler_%s%s_s%s.npy' %(ns,set_id,step))    
        
    euler_GSH = np.zeros([ns,H,el**3], dtype= 'complex128')
    
    for sn in range(ns):
        
        euler_GSH[sn,:,:] = gsh.GSH_Hexagonal_Triclinic(euler[sn,:,:])        
    
    np.save('euler_GSH_%s%s_s%s.npy' %(ns,set_id,step),euler_GSH)
    
    end = time.time()
    timeE = np.round((end - start),3)    
    
    msg = "Conversion from Euler angles to GSH coefficients completed: %s seconds" %timeE
    rr.WP(msg, wrt_file)
    